src/pages/BlockPage.vue

```diff
@@ -0,0 +1,424 @@
+        nonce() {
+            if (this.blockData) {
+                return this.blockData.nonce;
+            }
+            return '';
+        },
+        hash() {
+            if (this.blockData) {
+                return this.blockData.hash;
+            }
+            return '';
+        },
+        parentHash() {
+            if (this.blockData) {
+                return this.blockData.parentHash;
+            }
+            return '';
+        },
+    },
+    watch: {
+        '$route.params.block': {
+            handler(newBlock) {
+                if (this.block === newBlock) {
+                    return;
+                }
+                this.block = newBlock;
+                this.loadBlock();
+            },
+            immediate: true,
+        },
+        '$route.query.tab'(newTab) {
+            this.tab = newTab || defaultTab;
+        },
+        tab(newTab) {
+            this.$router.push({ query: { tab: newTab } });
+        },
+    },
+    methods: {
+        checkTabFromUrl() {
+            const tabQueryParam = this.$route.query.tab as string;
+            if (tabQueryParam && tabs.includes(tabQueryParam)) {
+                this.tab = tabQueryParam;
+            } else {
+                this.tab = defaultTab;
+            }
+        },
+        async loadBlock() {
+            try {
+                const blockResponse = await this.$indexerApi.get(`/block/${this.block}?includeAbi=true`);
+                if(blockResponse){
+                    this.blockData = blockResponse.data.results[0];
+                }
+            } catch (e) {
+                this.error = this.$t('components.failed_to_fetch_transactions');
+                console.error(e);
+            }
+        },
+        prevBlock() {
+            this.resetBlockData();
+            this.$router.push({ name: 'block', params: { block: parseInt(this.block) - 1 } });
+        },
+        nextBlock() {
+            this.resetBlockData();
+            this.$router.push({ name: 'block', params: { block: parseInt(this.block) + 1 } });
+        },
+        resetBlockData() {
+            this.blockData = null;
+        },
+    },
+});
+
+</script>
+<template>
+<div class="p-block">
+    <div class="p-block__header">
+        <span class="p-block__header-title">{{ $t('pages.blockpage.block') }}</span>
+        <span class="p-block__header-block-num">#{{ blockHeight }}</span>
+    </div>
+
+    <q-tabs
+        v-model="tab"
+        class="p-block__tabs-tabs text-blue shadow-2"
+        align="left"
+    >
+        <q-tab class="p-block__tabs-tab" name="overview" :label="$t('pages.blockpage.overview')" />
+        <q-tab class="p-block__tabs-tab" name="transactions" :label="$t('pages.blockpage.transactions')" />
+    </q-tabs>
+
+    <div class="p-block__main-container">
+        <div class="p-block__main-content">
+            <q-tab-panels v-model="tab" class="p-block__panels">
+                <q-tab-panel class="p-block__panel" name="overview">
```

Comment:
The content of this panel can be cleaned up by creating a new Component for the Block (in fact, two components, one for each card.

What do you think about creating a whole pair of components? is it worth it?

Reply:
yes, i think creating child components would be a good idea 👍  this one is a bit large as it is